eta/detectors/tfmodels/tfmodels_detectors.py

```diff
@@ -0,0 +1,144 @@
+'''
+Interface to the TensorFlow Models object detection library available at
+https://github.com/tensorflow/models/tree/master/research/object_detection.
+
+Copyright 2017-2019, Voxel51, Inc.
+voxel51.com
+
+Brian Moore, brian@voxel51.com
+'''
+# pragma pylint: disable=redefined-builtin
+# pragma pylint: disable=unused-wildcard-import
+# pragma pylint: disable=wildcard-import
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+from builtins import *
+# pragma pylint: enable=redefined-builtin
+# pragma pylint: enable=unused-wildcard-import
+# pragma pylint: enable=wildcard-import
+
+import os
+import sys
+
+import numpy as np
+import tensorflow as tf
+
+import eta.constants as etac
+from eta.core.config import Config
+from eta.core.geometry import BoundingBox, RelativePoint
+from eta.core.learning import ObjectDetector
+import eta.core.models as etam
+from eta.core.objects import DetectedObject, DetectedObjectContainer
+import eta.core.serial as etas
+import eta.core.utils as etau
+
+sys.path.append(os.path.join(etac.TF_OBJECT_DETECTION_DIR, "utils"))
+import label_map_util as gool
+
+
+# The default TFModelsDetectorConfig
+DEFAULT_CONFIG = os.path.join(
+    os.path.realpath(os.path.dirname(__file__)), "default-config.json")
+
+
+class TFModelsDetectorConfig(Config):
```

Comment:
Good question. This is a little bit subtle, but the definition here is intentional.

The design pattern I created is that All `ObjectDetector`s should be subclasses of `ObjectDetector`, but their associated `Config` classes should not subclass from `ObjectDetectorConfig`. This is because `ObjectDetectorConfig` is a meta-config that can load _any_ `ObjectDetector`. If you want to load a particular `ObjectDetector` instance, you create an `ObjectDetectorConfig` that specifies the `type` of the `ObjectDetector` and put its config in the `config` field. So the `Config` for a particular detector is just a regular `Config` subclass.